### PR TITLE
MAX32672: Updated AESKEYS and SYS Registers, Deprecated GPIO SetVSSEL and Pad Options.

### PR DESCRIPTION
**Pull request by sihyung-maxim**
Changes:
- Added two AES Keys (System and User) to ME21.
- Fixed SVD Workflow issues.
- Deprecated GPIO_SetVSSEL and pad options for ME21.
- Updated all register files with Doxygen additions.

### CODE REVIEW

**Comment by sihyung-maxim** [comment]
Making a change where all my future PRs will have the title with the chip names instead of the die names.

**Review by Jacob-Scheiffler — COMMENTED** [review]
Looks good to me. Probably should get some other approvals on this since I made some of these.

**Review by cwarer1 — APPROVED** [review]

File: Libraries/CMSIS/Device/Maxim/MAX32672/Include/aeskeys_regs.h
Comment (Jake-Carter) [inline Libraries/CMSIS/Device/Maxim/MAX32672/Include/aeskeys_regs.h:87]: I wonder if we can also clean up our Doxygen output for deprecated functions/structs.

Primarily we can add [@deprecated](https://www.doxygen.nl/manual/commands.html#cmddeprecated) to the header for the structs and register modules.

I'm also not sure exactly how the [current Doxygen deprecated list](https://analog-devices-msdk.github.io/msdk/Libraries/PeriphDrivers/Documentation/MAX32672/deprecated.html) is being generated.  There's not a `@deprecated` tag anywhere in the file.

File: Libraries/CMSIS/Device/Maxim/MAX32672/Include/max32672.h
Comment (Jake-Carter) [inline Libraries/CMSIS/Device/Maxim/MAX32672/Include/max32672.h:288]: Check if `@deprecated` works on macros as well?

File: Libraries/PeriphDrivers/Include/MAX32672/gpio.h
Comment (Jake-Carter) [inline Libraries/PeriphDrivers/Include/MAX32672/gpio.h:326]: Added the "API Change" label since we removed this function.  Before merging we should add a note about this to the PR.  It's an unsupported feature of the micro, but still good practice to highlight this and why we did it for the next release notes.

**Review by Jake-Carter — APPROVED** [review]
Looks good.  I think we could improve the Doxygen output before merging

File: Libraries/CMSIS/Device/Maxim/MAX32672/Include/aeskeys_regs.h
Comment (sihyung-maxim) [reply Libraries/CMSIS/Device/Maxim/MAX32672/Include/aeskeys_regs.h:87]: Ope, I didn't know @deprecated was actually used by Doxygen. I originally added the annotation into our deprecated register files, so the register workflows can filter out deprecated register files.

File: Libraries/CMSIS/Device/Maxim/MAX32672/Include/max32672.h
Comment (sihyung-maxim) [reply Libraries/CMSIS/Device/Maxim/MAX32672/Include/max32672.h:288]: It will throw an error if the c code sees the deprecated annotation. Will Doxygen know the macros are deprecated if you add a comment `// @deprecated` within the same line for each macro?

Comment (Jake-Carter) [reply Libraries/CMSIS/Device/Maxim/MAX32672/Include/max32672.h:288]: You should be able to add a Doxygen header for a preprocessor definition.  Though you may need the changes from #521 that enable Doxygen output for static members.  See [here](https://github.com/Jake-Carter/msdk/blob/0240928ae0077cb4651bbe52a4c526f63a71a2d0/Libraries/PeriphDrivers/Include/MAX32670/mxc_sys.h#L226) for an example where I did a doxygen header for a macro.

Ex:
```C
/*
 * @deprecated (description...)
 */
#define MXC_BASE_AESKEY MXC_BASE_AESKEYS
```

If you want to do it inline I think you have to do something like what we did for the enums.  Haven't tried this myself

```C
#define MXC_BASE_AESKEY MXC_BASE_AESKEYS /**< @deprecated (description) */
```

**Comment by Jake-Carter** [comment]
@sihyung-maxim looks like this did the trick:  https://github.com/Analog-Devices-MSDK/msdk/pull/534/commits/252ba1cd5ea5e4978d63b3fe31b65835cd692203

We also were missing some `@ingroup` tags.  Do you think we could get the register generators updated with those changes in that commit?  Essentially we need to make sure every `@file` tag and register module has the correct top-level `@ingroup`.



Deprecated list works now:
image

Modules are correctly listed and show their associated header files:

image

Header files show their members:

image

Source code can be browsed:

image

**Comment by Jake-Carter** [comment]
@sihyung-maxim also FYI I switched the drivers over to actually use the new header files:  https://github.com/Analog-Devices-MSDK/msdk/pull/534/commits/5cad636bf7514eb8b1d9afa123c94845be8cac3e

**Comment by Jake-Carter** [comment]
> @sihyung-maxim also FYI I switched the drivers over to actually use the new header files: [5cad636](https://github.com/Analog-Devices-MSDK/msdk/commit/5cad636bf7514eb8b1d9afa123c94845be8cac3e)

This broke pretty much every other micro's builds.  (see https://github.com/Analog-Devices-MSDK/msdk/actions/runs/4641783585)

We'll need to add some target checking in `aes_revb.c`.  I suggest:

```C
#if TARGET_NUM == 32672
#include "sys_aeskeys_regs.h"
#include "usr_aeskeys_regs.h"
#else
#include "aeskeys_regs.h"
#endif
```

File: Libraries/CMSIS/Device/Maxim/MAX32672/Include/usr_aeskeys_regs.h
Comment (sihyung-maxim) [inline Libraries/CMSIS/Device/Maxim/MAX32672/Include/usr_aeskeys_regs.h:80]: Should the @ingroup annotation for all the aeskeys be `aes`?

Comment (Jake-Carter) [reply Libraries/CMSIS/Device/Maxim/MAX32672/Include/usr_aeskeys_regs.h:80]: Any `@ingroup` will be listed like this:

image

So I think a good criteria for what gets the "root" `aes` group is to match exactly what is included by the peripheral's header file.

Ex "aes.h":
image

Then the Doxygen browser will also give you clickable links that match 1:1 with its source code.

image

**Review by Jake-Carter — APPROVED** [review]
Approved to merge.  We should apply these Doxygen improvements to the register file generator

**Review by Jake-Carter — CHANGES_REQUESTED** [review]
Sorry - scratch that.  I think we need to address https://github.com/Analog-Devices-MSDK/msdk/pull/534#issuecomment-1500694625

**Comment by sihyung-maxim** [comment]
> > @sihyung-maxim also FYI I switched the drivers over to actually use the new header files: [5cad636](https://github.com/Analog-Devices-MSDK/msdk/commit/5cad636bf7514eb8b1d9afa123c94845be8cac3e)
> 
> This broke pretty much every other micro's builds. (see https://github.com/Analog-Devices-MSDK/msdk/actions/runs/4641813965/jobs/8215147365)
> 
> We'll need to add some target checking in `aes_revb.c`. I suggest:
> 
> ```c
> #if TARGET_NUM == 32672
> #include "sys_aeskeys_regs.h"
> #include "usr_aeskeys_regs.h"
> #else
> #include "aeskeys_regs.h"
> #endif
> ```

Looks like the aes_revb.c doesn't even need the AESKEYS registers, so I've removed all aeskey_regs.h inclusions.

**Review by Jake-Carter — APPROVED** [review]
Approved to merge when checks pass

**Comment by Jake-Carter** [comment]
Manually running Build_Examples on this branch

https://github.com/Analog-Devices-MSDK/msdk/actions/runs/4877402221

**Comment by sihyung-maxim** [comment]
> @sihyung-maxim Approved to merge when checks pass. The diff on the register file is against my doxygen changes. Are those easy to integrate into the file generator?

Yep, should be implemented after this PR is merged. 

Took **way** to long to figure out that the BTM-CI doesn't let you run shell scripts in the way the SVD workflow previously had used. I've updated all (minus ME13B and ME55) register files to include your new doxygen changes. Should be good now.

**Review by Jacob-Scheiffler — APPROVED** [review]

**Pull request merged**
